### ISSUE TITLE
Http `Host` header can't be written properly into headers for backend request

### ISSUE THREAD

**Issue by zhao-kun** [megaease/easegress#27]
I want to set the `host` header of backend request, so I leverage `requestAdaptor` filter to set `Host`, the configuration is:
```
filters:
  - name: requestAdaptor
    kind: RequestAdaptor
    method: ""
    path: null
    header:
      del: []
      set:
        Host: "xxxx.com"
        X-Forwarded-Proto: "https"
```
but request sent to the backend with improper `host` header value (actually it's the backend IP address), this isn't fulfilling my expectation.

**Comment by benja-wu** [comment]
This behavior due to the Golang default implementation.[1]
I will update `requestAdpater` to allow users to set their wanted `HOST` value in the HTTP request header.

**Comment by benja-wu** [comment]
Merged by #28
